src/components/tree-views-container.js

```diff
@@ -433,6 +436,33 @@ const TreeViewsContainer = () => {
         [updateCurrentChildren, updateMapData]
     );
 
+    useEffect(() => {
+        let children = {};
+        let updated = false;
+        if (currentChildren != null) {
+            Object.values(currentChildren).map(
+                (e) => (children[e.elementName] = e)
+            );
+        }
+        Object.values(uploadingStudies)
+            .filter((e) => e.directory === selectedDirectory)
+            .forEach((e) => {
+                if (children[e.elementName] === undefined) {
+                    children[e.elementName] = e;
+                    updated = true;
+                }
+            });
+        if (updated) {
+            dispatch(
+                setCurrentChildren(
+                    Object.values(children).sort(function (a, b) {
+                        return a.elementName.localeCompare(b.elementName);
+                    })
+                )
+            );
+        }
+    }, [currentChildren, uploadingStudies, selectedDirectory, dispatch]);
```

Comment:
Maybye use a `currentChildrenRef = useRef()` here to avoid the currentChildren dependency which could become tricky because you call `dispatch(setCurrentChildren(` in this useEffect.

Reply:
done